infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/send/logreader/RoutingQueuesLogEntryReader.java

```diff
@@ -25,8 +35,29 @@ public RoutingQueuesLogEntryReader(CorfuRuntime runtime, LogReplicationSession s
 
     @Override
     public LogReplicationEntryMsg read(UUID logEntryRequestId) throws TrimmedException {
-        // Reads from the queue corresponding to this destination.  The queue contains addresses in the main queue
-        // from where the actual data(payload) is to be read.
         return null;
     }
+
+    @Override
+    protected OpaqueEntry filterTransactionEntry(OpaqueEntry opaqueEntry) {
+        List<SMREntry> routingTableEntryMsgs = opaqueEntry.getEntries()
+            .get(CorfuRuntime.getStreamID(LOG_ENTRY_SYNC_QUEUE_NAME_SENDER));
+
+        List<SMREntry> filteredMsgs = new ArrayList<>();
```

Comment:
actually do we really need this filtering? the opaque stream that this session's log entry reader is tracking is already tagged by the destination id